src/github.com/matrix-org/dendrite/roomserver/api/query.go

```diff
@@ -135,12 +136,12 @@ func NewRoomserverQueryAPIHTTP(roomserverURL string, httpClient *http.Client) Ro
 	if httpClient == nil {
 		httpClient = http.DefaultClient
 	}
-	return &httpRoomserverQueryAPI{roomserverURL, *httpClient}
+	return &httpRoomserverQueryAPI{roomserverURL, httpClient}
 }
 
 type httpRoomserverQueryAPI struct {
 	roomserverURL string
-	httpClient    http.Client
+	httpClient    *http.Client
```

Comment:
I'm changing it because otherwise we are copying the http.Client by value. Passing an http.Client around by value is mosty safe because the http.Client is actually just a pointer to a http.RoundTripper and some config. However if the user changes the config *after* passing the http.Client to NewRoomserverQueryAPI then we'll end up in a weird state.

(We have been using the default client everywhere, but that's just because we haven't got round to setting up a custom client in any of the components yet.)